app/controllers/stats_controller.rb

```diff
@@ -5,7 +5,7 @@ def subscriptions
       @tags[tag.tagname] = @tags[tag.tagname] || 0
       @tags[tag.tagname] += 1
     end
-    render plain: @tags.inspect, status: 200
+    @tags = @tags.group_by { |_k, v| v }.map { |k, v| { k => v.map { |x| x.join("-") } } }
```

Comment:
So let's take this for example:

```ruby
> tags
=> {"boagd"=>3, "boa"=>34, "boag"=>39, "b"=>66}
```

see how if group by v/10 then 34 becomes 3, 66 becomes 6, so ones within 10 of each other get added to the same group?

```ruby
> tags.group_by { |k, v| v/10 }
=> {0=>[["boagd", 3]], 3=>[["boa", 34], ["boag", 39]], 6=>[["b", 66]]}
```

I think the second `map` is something we can do in the template though, since it relates to how we want to display the data. So i'll suggest:

```suggestion
    @tags = @tags.group_by { |k, v| v/10 }
    @tags = @tags.reverse # because it's naturally sorted "ascending", so let's flip it!
```

Reply:
nice that the group is automatically sorted, too!

Reply:
Yes actually this works better and is way cleaner. The ruby hash doesn't have the reverse method though I think it's safe to use `sort`?